File: src/baseDatos/Serializador.java
package baseDatos;

import java.io.*;
import java.io.Serializable;

public class Serializador {
   
        public static void serializar(Serializable objeto, String strArchivo){

            ObjectOutputStream objectOutputStream;
            FileOutputStream fileOutputStream;

            try{
                fileOutputStream
                = new FileOutputStream(strArchivo);
   
                 objectOutputStream 
                = new ObjectOutputStream(fileOutputStream);       
   
            objectOutputStream.writeObject(objeto);
            objectOutputStream.flush();
            objectOutputStream.close();
   
            }catch(IOException e){

                System.out.println("ERROR: HA OCURRIDO UN ERROR EN LA SERIALIZACIÓN");
                e.printStackTrace();

            }
        }

        public static void guardarFacturas(){

            serializar(Load.facturas,
             "src/baseDatos/temp/facturas.txt");

            }

        public static void guardarTiendas(){

            serializar(Load.tiendas, "src/baseDatos/temp/tiendas.txt");

        }

        public static void guardarFabrica(){

            serializar(Load.fabrica, "src/baseDatos/temp/fabrica.txt");
        }

        public static void guardarCatalogo(){
            serializar(Load.catalogo, "src/baseDatos/temp/catalogo.txt");
        }

        public static void guardarClientes(){
            serializar(Load.clientes, "src/baseDatos/temp/clientes.txt");
        }

         public static void guardarVendedores(){
            serializar(Load.vendedores, "src/baseDatos/temp/vendedores.txt");
         }

         public static void guardarTransporte(){
             serializar(Load.transporteAbastecer, "src/baseDatos/temp/transporte.txt");
         }

         public static void guardarConductores(){
            serializar(Load.conductores, "src/baseDatos/temp/conductores.txt");
         }

         public static void guardarAtributos(){
            serializar(Load.infoAtributos, "src/baseDatos/temp/infoAtributos.txt");
         }
        
       
}

File: Python/src/baseDatos/Serializador.py
import pickle
from ..gestorAplicacion.gestion.Operario import Operario
from ..gestorAplicacion.gestion.Vendedor import Vendedor
from ..gestorAplicacion.produccion.Fabrica import Fabrica
from ..gestorAplicacion.produccion.Producto import Producto
from ..gestorAplicacion.produccion.Tienda import Tienda
from ..gestorAplicacion.gestion.Factura import Factura
from ..gestorAplicacion.gestion.Cliente import Cliente
from ..gestorAplicacion.gestion.Conductor import Conductor
from ..gestorAplicacion.produccion.Transporte import Transporte

class Serializador:

    @classmethod
    def serializar():

        #Fábrica
        pickleFileFabrica = open("src/baseDatos/temp/pickleFileFabrica.pkl","wb")
        pickle.dump(Fabrica.getListaFabricas(),pickleFileFabrica)
        pickleFileFabrica.close()

        #Tiendas
        pickleFileTienda = open("src/baseDatos/temp/pickleFileTienda.pkl","wb")
        pickle.dump(Tienda.getListaTiendas(),pickleFileTienda)
        pickleFileTienda.close()

        #Productos
        pickleFileProducto = open("src/baseDatos/temp/pickleFileProducto.pkl","wb")
        pickle.dump(Producto.getListaProductos(),pickleFileProducto)
        pickleFileProducto.close()

        #Facturas
        pickleFileFactura = open("src/baseDatos/temp/pickleFileFactura.pkl","wb")
        pickle.dump(Factura.getListaFacturas(),pickleFileFactura)
        pickleFileFactura.close()

        #Clientes
        pickleFileCliente = open("src/baseDatos/temp/pickleFileCliente.pkl","wb")
        pickle.dump(Cliente.getListaClientes(),pickleFileCliente)
        pickleFileCliente.close()

        #Vendedores
        pickleFileVendedor = open("src/baseDatos/temp/pickleFileVendedor.pkl","wb")
        pickle.dump(Vendedor.getListaVendedores(),pickleFileVendedor)
        pickleFileVendedor.close()

        #Conductores
        pickleFileConductor = open("src/baseDatos/temp/pickleFileConductor.pkl","wb")
        pickle.dump(Conductor.getListaConductores(),pickleFileConductor)
        pickleFileConductor.close()

        #Operarios
        pickleFileOperario = open("src/baseDatos/temp/pickleFileOperario.pkl","wb")
        pickle.dump(Operario.getListaOperarios(),pickleFileOperario)
        pickleFileOperario.close()

        #Transportes
        pickleFileTransporte = open("src/baseDatos/temp/pickleFileTransporte.pkl","wb")
        pickle.dump(Transporte.getListaTransportes(),pickleFileTransporte)
        pickleFileTransporte.close()